flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/ExecutorImpl.java

```diff
@@ -577,4 +593,29 @@ private void closeSession() throws SqlExecutionException {
             // ignore any throwable to keep the cleanup running
         }
     }
+
+    private static Collection<HttpHeader> readHeadersFromEnvironmentVariable(String envVarName) {
+        List<HttpHeader> headers = new ArrayList<>();
+        String rawHeaders = System.getenv(envVarName);
+
+        if (rawHeaders != null) {
+            String[] lines = rawHeaders.split("\n");
```

Comment:
Do you think it will be not so easy to specify a single env variable value that is separated by newlines? What about considering a different separator for distinguishing a list of header key-value pairs in a single line?

Reply:
nit: instead of `\n`, should we consider platform independent `System.lineSeparator()`?

Reply:
I believe this is actually the best option because, as RFC 7230 states, CRLF within header values was only expected as part of "obsolete folding", which [is not supported anymore](https://datatracker.ietf.org/doc/html/rfc7230#:~:text=Historically%2C%20HTTP%20header%20field%20values%20could%20be%20extended%20over%0A%20%20%20multiple%20lines).

```
     header-field   = field-name ":" OWS field-value OWS

     field-name     = token
     field-value    = *( field-content / obs-fold )
     field-content  = field-vchar [ 1*( SP / HTAB ) field-vchar ]
     field-vchar    = VCHAR / obs-text

     obs-fold       = CRLF 1*( SP / HTAB )
                    ; obsolete line folding
                    ; see [Section 3.2.4](https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.4)
```
field-value, on the other hand rules out anything in VCHAR, so  expecting a delimiter and an escaping strategy for a character that can naturally occur in a cookie is not user-friendly.

Reply:
HTTP header-wise it seems ok, I thought it will be challenging for specifying the env variable with multi-line values (each separated with a new line. this is just a nit comment.